packages/db/src/models/base.ts

```diff
@@ -96,6 +96,26 @@ class BasePgModel<ItemType, RecordType extends { cumulus_id: number }> {
     return record.cumulus_id;
   }
 
+  /**
+   * Get cumulus_id column value for multiple records in Postgres
+   *
+   * @param {Knex | Knex.Transaction} knexOrTransaction -
+   *  DB client or transaction
+   * @param {Array<string>} columnNames - column names for whereIn query
+   * @param {Knex.QueryCallback} values - record values for whereIn query
+   * @returns {Promise<Array<number>>} An array of cumulus_ids for the returned records
+   */
+  async getRecordsCumulusIds(
+    knexOrTransaction: Knex | Knex.Transaction,
+    columnNames: Array<string>,
```

Comment:
```suggestion
columnNames: Array<keyof Partial<RecordType>>
```

This will give us some type safety of ensuring that the column names have to be valid column names for the record type

Reply:
This still needs to be addressed